app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -35,6 +37,42 @@ def events
       respond_for_index :collection => collection
     end
 
+    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
```

Comment:
The most concrete example I have is the "splice" case where data from an external source is being populated into katello. All aspects (full CRUD) of the content host need to be made available for this, including setting custom registration and check-in times, for example.

As I've said, the rhsm api is not expected to be a user endpoint, it is a tooling endpoint. hammer commands certainly shouldn't hook into that.

I don't really understand the pushback against having a supported full CRUD for all aspects of a host. Even if I said that it's simply for testing that seems like a legitimate value add. As it is, there are several existing tools that need to continue to be supported post host-unification.

We deprecated an existing api that needs to get a replacement. Pretty please.